src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java

```diff
@@ -87,35 +90,25 @@ of this software and associated documentation files (the "Software"), to deal
      */
     @SuppressFBWarnings(value="DC_DOUBLECHECK")
     public static DatadogClient getInstance(String url, String logIntakeUrl, Secret apiKey){
-        if(enableValidations){
-            if (url == null || url.isEmpty()) {
-                logger.severe("Datadog Target URL is not set properly");
-                return null;
-            }
-            if (apiKey == null || Secret.toString(apiKey).isEmpty()){
-                logger.severe("Datadog API Key is not set properly");
-                return null;
-            }
-            if (logIntakeUrl == null || logIntakeUrl.isEmpty()){
-                logger.warning("Datadog Log Intake URL is not set properly");
-            }
-        }
-
-        if(instance == null){
-            synchronized (DatadogHttpClient.class) {
-                if(instance == null){
-                    instance = new DatadogHttpClient(url, logIntakeUrl, apiKey);
+        // If the configuration has not changed, return the current instance without validation
+        // since we've already validated and/or errored about the data
+        if (lastInstance != null && (StringUtils.equals(lastInstance.getLogIntakeUrl(), logIntakeUrl))
+        && (StringUtils.equals(lastInstance.getUrl(), url)
+        && ((lastInstance.getApiKey() == null && apiKey == null) || lastInstance.getApiKey().equals(apiKey)))){
+            return instance;
+        } else {
+            DatadogHttpClient.lastInstance = new DatadogHttpClient(url, logIntakeUrl, apiKey);
+            if (enableValidations) {
+                try {
+                    validateCongiguration(url, logIntakeUrl, apiKey);
```

Comment:
Typo + `validateConfiguration` should not be a static method but instead you should do `newInstance.validateConfiguration()`

Reply:
Good catch!

Reply:
Reopening: this should not be a static method.